ValheimExportHelper/Resources/RestoreShadersOnSceneLoad.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using System.IO.Compression;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+
+[InitializeOnLoad]
+public class RestoreShadersOnSceneLoad : MonoBehaviour
+{
+    public static void RestoreOriginalShaders() {
+        if (Directory.Exists("Assets/Shader"))
+        {
+            if(Directory.EnumerateFiles("Assets/Shader").ToList().Count >=1)
+            {
+                Directory.Delete("Assets/Shader", recursive: true);
+            }
+        }
+        
+        var zip = ZipStorer.Open("Assets"+ Path.DirectorySeparatorChar +"Shader_Original.zip", FileAccess.Read);
+        List<ZipStorer.ZipFileEntry> dir = zip.ReadCentralDir();
+        foreach (ZipStorer.ZipFileEntry entry in dir)
+        {
+                string strExeFilePath = System.Reflection.Assembly.GetExecutingAssembly().Location;
+                string strWorkPath = Path.GetDirectoryName(strExeFilePath);
+                char[] trimcharacters =
+                {
+                     'S', 'c', 'r', 'i', 'p', 't', 'A', 's', 's', 'e', 'm', 'b', 'l', 'i', 'e', 's'
+                };
+                char[] trimchar2 =
+                {
+                    'L', 'i', 'b', 'r', 'a', 'r', 'y'
+                };
+                
+                string finalPath = strWorkPath.TrimEnd(trimcharacters);
+                string correctedPath = finalPath.TrimEnd(Path.DirectorySeparatorChar);
+                string correctedPath1 = correctedPath.TrimEnd(trimchar2);
+                string forrealPath = correctedPath1  + "Assets" + Path.DirectorySeparatorChar;
+                zip.ExtractFile(entry, forrealPath + entry.FilenameInZip);

```

Comment:
...why, what is the relative path?

Also this trims any character from the given set of chars, not the characters in that order.

Reply:
TBH I couldnt figure out why the lib wasn't just using relative path for extracting. 

NET4.0 limits us to not use the zipfile function that is available in 4.5  so I had to go with an older known solution

I found it to be causing some REALLY odd errors at random on scene switch/load/startup when attempting to run the zipfile function

So that was why I ventured down this path.

Reply:
Also this dumps to the Assets directory? Just use "../../Assets/Shader/".

Reply:
Which in retrospect is probably a poor solution by design and should be scrapped for some kind of static extension that can be reached within the scripts folder in order to utilize the ZipFile method from .Net 4.5

Reply:
So apparently ZipFile is in .NET 4.0 but the assembly isn't referenced by default.

Reply:
> So apparently ZipFile is in .NET 4.0 but the assembly isn't referenced by default.

I was having a heck of a time finding any kind of documentation of a non lib assisted zip/unzip method for .NET 4.0